cmds/uinit/uinit.go

```diff
@@ -6,35 +6,145 @@
 package main
 
 import (
+	"fmt"
+	"io/ioutil"
 	"log"
 	"os"
 	"os/exec"
 	"strings"
+	"syscall"
+
+	"github.com/u-root/u-root/uroot"
 )
 
 var (
 	commands = []string{
-		"/buildbin/date",
-		"/buildbin/ls -l",
+		"/bin/bash",
 	}
+
+	DEFAULT_ROOT_DEV = "/dev/mmcblk0p1"
+	DEVICE_PARAM     = "uroot.rootdevice"
 )
 
-func main() {
-	for _, line := range commands {
-		log.Printf("Executing Command: %v", line)
-		cmdSplit := strings.Split(line, " ")
-		if len(cmdSplit) == 0 {
-			continue
-		}
+func exec_command(path string) error {
+	// Will exec and dup a command at path
+	cmd := exec.Command(path)
+	cmd.Stdin = os.Stdin
+	cmd.Stderr = os.Stderr
+	cmd.Stdout = os.Stdout
+	var fd int
+	cmd.SysProcAttr = &syscall.SysProcAttr{Ctty: fd, Setctty: true, Setsid: true, Cloneflags: uintptr(0)}
```

Comment:
You can make this
```
cmd.SysProcAttr = &syscall.SysProcAttr{
  Setctty: true,
  Setsid: true,
}
```

reason: by default, structure values will be initialized to 0. fd will be 0. So no need to set Ctty and Cloneflags.

as a side note, you also need not cast constants. `Cloneflags: 0` is sufficient.